fili-core/src/test/groovy/com/yahoo/bard/webservice/druid/client/impl/AsyncDruidWebServiceImplSpec.groovy

```diff
@@ -0,0 +1,53 @@
+package com.yahoo.bard.webservice.druid.client.impl
+
+import com.yahoo.bard.webservice.druid.client.DruidClientConfigHelper
+import com.yahoo.bard.webservice.druid.model.query.QueryContext
+import com.yahoo.bard.webservice.druid.model.query.WeightEvaluationQuery
+
+import com.fasterxml.jackson.databind.ObjectMapper
+
+import io.netty.handler.codec.http.HttpHeaders
+import spock.lang.Specification
+
+import java.util.function.Supplier
+
+class AsyncDruidWebServiceImplSpec extends Specification {
+    def "Ensure that headersToAppend are added"() {
+        setup:
+        WeightEvaluationQuery weightEvaluationQuery = Mock(WeightEvaluationQuery)
+        QueryContext queryContext = Mock(QueryContext)
+        weightEvaluationQuery.getContext() >> { queryContext }
+        queryContext.numberOfQueries() >> { 1 }
+        queryContext.getSequenceNumber() >> { 1 }
+
+        when:
+        Map<String, String> expectedHeaders = new HashMap<>()
+        expectedHeaders.put("k1", "v1")
+        expectedHeaders.put("k2", "v2")
+        Supplier<Map<String, String>> supplier = new Supplier<Map<String, String>>() {
+            @Override
+            Map<String, String> get() {
+                return expectedHeaders
+            }
+        }
+
+        AsyncDruidWebServiceImplWrapper webServiceImplWrapper = new AsyncDruidWebServiceImplWrapper(
+                DruidClientConfigHelper.getNonUiServiceConfig(),
+                new ObjectMapper(),
+                supplier
+        )
+        webServiceImplWrapper.postDruidQuery(
+                null,
+                null,
+                null,
+                null,
+                weightEvaluationQuery
+        );
+
+        then:
+        HttpHeaders actualHeaders = webServiceImplWrapper.getHeaders()
+        for (Map.Entry<String, String> header : expectedHeaders) {
+            actualHeaders.get(header.getKey()) == header.getValue()
```

Comment:
Only top-level expressions in a `then` block are treated as assertions. To treat non-top-level expressions as assertions, you need to explicitly add the `assert` keyword: `assert actualHeaders.get(header.getKey()) == header.getValue()`